LoRaEngine/modules/LoRaWanNetworkSrvModule/LoraTools/Regions/RegionLimits.cs

```diff
@@ -13,24 +13,15 @@ public class RegionLimits
         /// </summary>
         public (double min, double max) FrequencyRange { get; set; }
 
-        public List<string> DatarateRange { get; set; }
+        public HashSet<string> DownstreamValidDR { get; set; }
 
-        public bool IsCurrentDRIndexWithinAcceptableValue(uint dr)
-        {
-            if (dr < this.DatarateRange.Count)
-            {
-                return true;
-            }
-            else
-            {
-                return false;
-            }
-        }
+        public HashSet<string> UpstreamValidDR { get; set; }
```

Comment:
If you end up adding methods to lookup in here, make these private. Having them exposed as public is opening up direct manipulation to it. Keeping them private helps with protecting your state.